http-clients/netty-nio-client/src/test/java/software/amazon/awssdk/http/nio/netty/internal/nrs/HandlerSubscriberTest.java

```diff
@@ -186,6 +194,7 @@ public boolean awaitTermination(long timeout, TimeUnit unit) {
 
         @Override
         public ChannelFuture register(Channel channel) {
+            ((CustomEmbeddedChannel) channel).loop = this;
```

Comment:
Needed to support the change in behavior introduced by https://github.com/netty/netty/commit/ef1f637b0f9979d98dbb0f3e5281d7272012c76d:

```
java.lang.ClassCastException: software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerSubscriberTest$CustomEmbeddedEventLoop cannot be cast to io.netty.channel.embedded.EmbeddedEventLoop
--
at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerSubscriberTest.teardown(HandlerSubscriberTest.java:69)
 
[System Output]
2021-12-15 16:41:35,290 [main] WARN  io.netty.channel.embedded.EmbeddedChannel - More than one exception was raised. Will report only the first one and log others.
java.lang.ClassCastException: software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerSubscriberTest$CustomEmbeddedEventLoop cannot be cast to io.netty.channel.embedded.EmbeddedEventLoop
at io.netty.channel.embedded.EmbeddedChannel.embeddedEventLoop(EmbeddedChannel.java:680)
at io.netty.channel.embedded.EmbeddedChannel.runPendingTasks(EmbeddedChannel.java:602)
at io.netty.channel.embedded.EmbeddedChannel$EmbeddedUnsafe$1.beginRead(EmbeddedChannel.java:842)
at io.netty.channel.DefaultChannelPipeline$HeadContext.read(DefaultChannelPipeline.java:1362)
at io.netty.channel.AbstractChannelHandlerContext.invokeRead(AbstractChannelHandlerContext.java:686)
at io.netty.channel.AbstractChannelHandlerContext.read(AbstractChannelHandlerContext.java:671)
at io.netty.channel.DefaultChannelPipeline.read(DefaultChannelPipeline.java:1004)
at io.netty.channel.AbstractChannel.read(AbstractChannel.java:290)
at io.netty.channel.DefaultChannelPipeline$HeadContext.readIfIsAutoRead(DefaultChannelPipeline.java:1422)
at io.netty.channel.DefaultChannelPipeline$HeadContext.channelActive(DefaultChannelPipeline.java:1400)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelActive(AbstractChannelHandlerContext.java:230)
at io.netty.channel.AbstractChannelHandlerContext.invokeChannelActive(AbstractChannelHandlerContext.java:216)
at io.netty.channel.DefaultChannelPipeline.fireChannelActive(DefaultChannelPipeline.java:895)
at io.netty.channel.AbstractChannel$AbstractUnsafe.register0(AbstractChannel.java:522)
at io.netty.channel.AbstractChannel$AbstractUnsafe.register(AbstractChannel.java:480)
at io.netty.channel.embedded.EmbeddedChannel$EmbeddedUnsafe$1.register(EmbeddedChannel.java:799)
at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerSubscriberTest$CustomEmbeddedEventLoop.register(HandlerSubscriberTest.java:195)
at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerSubscriberTest$CustomEmbeddedEventLoop.register(HandlerSubscriberTest.java:189)
at software.amazon.awssdk.http.nio.netty.internal.nrs.HandlerSubscriberTest.setup(HandlerSubscriberTest.java:61)
```